content/en/blog/2022/knative.md

```diff
@@ -0,0 +1,123 @@
+
+Now let's take a look at an example trace with two workloads (first and second). The workflow is similar to the diagram from the previous section: the first service receives an HTTP call and sends a cloud event to the second service. The full demo source code can be found in [pavolloffay/knative-tracing](https://github.com/pavolloffay/knative-tracing).
+
+![A screenshot from Jaeger that shows Knative trace](/img/blog-knative/jaeger-knative-trace.jpg)
+
+The trace shows the following services interacting: activator, first workload, broker-ingress, imc-dispatcher, broker-filter, activator, and second workload. There are many services, right? A simple interaction of two workloads resulted in a trace that shows many Knative internal components. From the observability perspective, this is great because it can show issues in the infrastructure and additionally show cost associated with Knative request processing.
+
+Let's briefly example the data flow. The incoming HTTP request first goes through an activator service that is responsible for scaling up a workload, then its execution reaches the first workload. The first workload sends a cloud event which goes through the broker and dispatcher and finally reaches the second workload.
+
+Now let's take a closer look at the user workloads. The first service is a Golang service with a single REST API endpoint. The endpoint implementation creates a cloud event and sends it to the broker. Let's take a look at important facts from the observability perspective:
+* REST API is instrumented with OpenTelemetry. This allows us to link traces started in the Knative activator service with spans created in the workload and further link it with outbound spans - e.g. to calls to the second service.
+* The workload is using instrumented [Cloudevents client/SDK](https://github.com/cloudevents/sdk-go/tree/main/observability/opentelemetry/v2) - his is related to the previous point and it allows us to continue the trace in the outbound request - in our case to the second service.
+
+How is the trace-context (`traceId`, `spanId`, `sampled` flag) being propagated in our example applications? The trace-context is propagated in HTTP headers both for incoming HTTP requests into the first service and as well for cloud events sent to the second service. The trace-context is not attached directly to the event extensions/attributes.
+
+Follows log output with request headers from the first service:
+```bash
```

Comment:
this is a lot of headers with identification of a request. Curiuos, where is the `X-Request-Id` coming from?

Reply:
Honestly I don't know, I can ask around and get back to you